code/modules/clothing/glasses/engine_goggles.dm

```diff
@@ -117,14 +119,17 @@
 		flick_overlay(pic, list(user.client), 8)
 
 /obj/item/clothing/glasses/meson/engine/update_icon()
-	icon_state = "[initial(icon_state)]-[mode]"
+	icon_state = "trayson-[mode]"
```

Comment:
I'd honestly prefer we keep the initial(icon_state) so subtypes can have potentially different sprites for when it's just a subset of modes

Reply:
I prefer it as well, however maintaining the old behavior while fixing the issues with the chameleons would require 
a) changing the base icon_state of each item to the on sprite to prevent ez meta 
a.1) naming the sprites out of order to facilitate this (ie on would just be trayson, off would be trayson-off, ect)
b) duplicating all 12 sprites for each child, even if nothing is different beside the name of the icon_state

Reply:
Made it a bit easier to override the update_icon without having to copypasta all the same code.